src/js/components/Anchor/Anchor.js

```diff
@@ -105,6 +107,7 @@ const Anchor = forwardRef(
           if (onBlur) onBlur(event);
         }}
         size={sizeProp || size}
+        {...(withinThemeContext === undefined ? { theme } : {})}
```

Comment:
This pattern is repeated way too much and is awkward so it's intent is hard to understand.

Also, if a `theme` prop is passed in, doesn't it end up in `rest` and get overwritten by the theme from the context if there is one?

I think a pattern of something like:
```jsx
const theme = useTheme(themeProp);

<StyledAnchor theme={theme} ... />
```
Is much cleaner assuming the `useTheme` does the `useContext(ThemeContext)` and logic to reconcile the theme prop from the component and theme from the theme context if needed.

This also could be done to prevent the theme parameter from changing each render (your ternary create two new objects each render)

Reply:
> Also, if a theme prop is passed in, doesn't it end up in rest and get overwritten by the theme from the context if there is one?

I tried to keep `{...(withinThemeContext === undefined ? { theme } : {})}` above `{...rest}` so that rest would override the theme, looks like I actually missed this on Anchor though.

> I think a pattern of something like:
> 
> const theme = useTheme(themeProp);
> 
> <StyledAnchor theme={theme} ... />
> 
> Is much cleaner assuming the useTheme does the useContext(ThemeContext) and logic to reconcile the theme prop from the component and theme from the theme context if needed.

I wanted to avoid having an explicit `themeProp` on each component and passing through the theme on each styled component if we don't need it (if we are within a theme context already). I'm not sure performance wise if it matters or not if we pass a large object (the theme) as a prop for each styled component regardless of whether we are within a theme context or not.

> This also could be done to prevent the theme parameter from changing each render (your ternary create two new objects each render)

This is a good point, I hadn't thought about that cost

I'll add that performance is an area I'm not super familiar with so definitely open to other perspectives and open to going this route, just wanted to share my thought process of why I approached this the way that I did.

Reply:
I'm not sure what you're concerned with as far as performance of passing a theme prop. It seems like you're already doing that in most of the cases via the value from useThemeValue and passing it to the styled component (like Anchor here.) I would think in most cases you'd get the benefit with no real cost if `useThemeValue()` accepted a themeProp arg and did the logic discussed.  Components that use useThemeValue and others could easily just check their props for the theme prop and pass it along. I don't think passing a large object is expensive (ie. it's a reference unless you're doing something expensive like a merge.)

Reply:
Ok sounds good, I'll work on making this change

Reply:
I implemented this change but I'm seeing slightly slower performance on the com repo. Maybe if someone is willing they could run on their end as well just to validate that they are getting the similar results as I am. I'll push up the changes in a new branch so it is easier to compare.

Here is what I'm seeing with this approach:
<img width="749" alt="Screenshot 2024-09-05 at 10 32 13 AM" src="https://github.com/user-attachments/assets/2c917627-2126-41bc-bcd9-dc857ad55e81">
<img width="738" alt="Screenshot 2024-09-05 at 10 32 31 AM" src="https://github.com/user-attachments/assets/0e1591ad-2e1b-4602-aac2-1c8795aa1fc3">
<img width="685" alt="Screenshot 2024-09-05 at 11 02 59 AM" src="https://github.com/user-attachments/assets/f8ea678f-5588-4929-92ed-d2086f582093">

Reply:
I am testing from a different location than yesterday so my internet speed might be a little worse

Reply:
Branch with the alternate approach https://github.com/grommet/grommet/pull/7306

Reply:
I just tested this branch again to see if me being in a different location on different internet was causing a problem, it looks like it is not. Here is what I got
<img width="471" alt="Screenshot 2024-09-05 at 11 19 01 AM" src="https://github.com/user-attachments/assets/15cb0cb6-f843-466e-82cb-11afe5175e33">
<img width="430" alt="Screenshot 2024-09-05 at 11 20 09 AM" src="https://github.com/user-attachments/assets/e8420f00-e018-4b9f-bb2c-93bdb193182b">

Reply:
To summarize the results that I am getting:

This branch with `withinThemeContext` approach = ~1500

Alternate branch https://github.com/grommet/grommet/pull/7306 with `themeProp` approach = ~2500

Again would be good to have someone else validate these results just to make sure we are seeing the same thing

Reply:
> I just tested this current branch again to see if me being in a different location on different internet was causing a problem, it looks like it is not. Here is what I got <img alt="Screenshot 2024-09-05 at 11 19 01 AM" width="471" src="https://private-user-images.githubusercontent.com/54560994/364882162-15cb0cb6-f843-466e-82cb-11afe5175e33.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU1NjA2NjcsIm5iZiI6MTcyNTU2MDM2NywicGF0aCI6Ii81NDU2MDk5NC8zNjQ4ODIxNjItMTVjYjBjYjYtZjg0My00NjZlLTgyY2ItMTFhZmU1MTc1ZTMzLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTA1VDE4MTkyN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPWQ3NWJlMzY0YWEyZjg1OWY3MTMxZDNmNTA2MmI5NjY1ZjI4NThjMmU1MTI4N2ZhYmFmNjI1NTFlYTliNWJmNGQmWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.kUViPBD0AsUnx_XxKaJ6fWhLthwcdicVb9eNoC9aQoA"> <img alt="Screenshot 2024-09-05 at 11 20 09 AM" width="430" src="https://private-user-images.githubusercontent.com/54560994/364882329-e8420f00-e018-4b9f-bb2c-93bdb193182b.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MjU1NjA2NjcsIm5iZiI6MTcyNTU2MDM2NywicGF0aCI6Ii81NDU2MDk5NC8zNjQ4ODIzMjktZTg0MjBmMDAtZTAxOC00YjlmLWJiMmMtOTNiZGIxOTMxODJiLnBuZz9YLUFtei1BbGdvcml0aG09QVdTNC1ITUFDLVNIQTI1NiZYLUFtei1DcmVkZW50aWFsPUFLSUFWQ09EWUxTQTUzUFFLNFpBJTJGMjAyNDA5MDUlMkZ1cy1lYXN0LTElMkZzMyUyRmF3czRfcmVxdWVzdCZYLUFtei1EYXRlPTIwMjQwOTA1VDE4MTkyN1omWC1BbXotRXhwaXJlcz0zMDAmWC1BbXotU2lnbmF0dXJlPTg1ZWQyN2FjYmExY2FiZTM4NjBlZDk1YzFhZjg0NGUxN2E2ZTI1NzVlMjRmNzMzNWNlZDM2YjdlYTNjZWQwMzImWC1BbXotU2lnbmVkSGVhZGVycz1ob3N0JmFjdG9yX2lkPTAma2V5X2lkPTAmcmVwb19pZD0wIn0.jSIVPLENHehy51Qs8vQ_G5dUkBC9XcWrQkPhr5xs9w0">

Can you please label these images with product + approach/version so that we know what we are comparing? Also, we need the equivalent for v2.38 so that we have our baseline.

Reply:
Yeah I'll add more detailed labels, baseline can be found in the PR description

Reply:
Com, grommet 2.38.0 tested at new location
<img width="407" alt="Screenshot 2024-09-05 at 1 49 00 PM" src="https://github.com/user-attachments/assets/00c8484e-1900-40a3-8c31-86a368c2fb03">